errorDashboardJS/src/apiClient.ts

```diff
@@ -0,0 +1,107 @@
+import { customFetch } from "./fetch";
+import { configs, type Configs } from "./configs";
+import { baseUrl } from "./environment";
+import type { CreateErrorRequestType, CreateTagRequestType } from "./types";
+
+interface InitializeClient {
+  clientId: string;
+  clientSecret: string;
+  window: Window;
```

Comment:
imo don't have them pass you `window`. Just use it. I know there are some edge-cases here, but I wouldn't sweat it this early.

Also, you only use it to derive `userAgent`, which imo isn't worth adding _yet_ (maybe in a follow-up PR once you have the core code finished), so maybe just remove `window` entirely for now.